added public pauseAnimation and resumeAnimation features

yiziz opened the pull request:
Some trailing white space have also been trimmed by my Atom editor.  This addresses the issue ( #15 ) I opened.

I also added fingerprinting for the min.js file.

yiziz committed:
added public pauseAnimation and resumeAnimation features

yiziz committed:
added fingerprinting to build task

marcbruederlin commented:
Hey yiziz, thanks for your pull request. I will check this in the next few days.

yiziz committed:
made render a bit more efficient with variable caching and batch particle filling

yiziz committed:
updated README to document showParticles option

yiziz commented:
Hey, this pull request kind of got out of hand since I just kept working and added a few more things.

To summarize:

- I added public pauseAnimation and resumeAnimation functions.  The related commit also erased some trailing white space due to my Atom editor
- I added fingerprinting to the min.js file by using `gulp-rev`
- I refactored a few things to [batch render particles](https://stackoverflow.com/a/31737509) in the canvas, [reduced the number of edge distance calculations](https://github.com/marcbruederlin/particles.js/pull/16/commits/0b355588eba45b9c361ba2561a1cf6b9490be6e6#diff-6d581ad56a8c69e7873b92103bd920eeR354), and did a little bit of variable caching
- I updated the `README` to document the added `showParticles` option

I tried to keep to your coding style as much as I could, but I normally don't use `;` and `_` in my JavaScript.  Hopefully everything is good.
